src/seedsigner/models/seed_storage.py

```diff
@@ -83,23 +74,17 @@ def get_pending_mnemonic_word(self, index: int) -> str:
 
     def get_pending_mnemonic_fingerprint(self, network: str = SettingsConstants.MAINNET) -> str:
         try:
-            if self._pending_is_electrum:
-                seed = ElectrumSeed(self._pending_mnemonic)
-            else:
-                seed = Seed(self._pending_mnemonic)
+            seed = self._Seed_cls(self._pending_mnemonic)
             return seed.get_fingerprint(network)
         except InvalidSeedException:
             return None
 
 
     def convert_pending_mnemonic_to_pending_seed(self):
-        if self._pending_is_electrum:
-            self.pending_seed = ElectrumSeed(self._pending_mnemonic)
-        else:
-            self.pending_seed = Seed(self._pending_mnemonic)
+        self.pending_seed = self._Seed_cls(self._pending_mnemonic)
         self.discard_pending_mnemonic()
     
 
     def discard_pending_mnemonic(self):
         self._pending_mnemonic = []
-        self._pending_is_electrum = False
+        self._Seed_cls = Seed
```

Comment:
I'm not sure this reset is necessary nor the default value in `__init__()`. If `init_pending_mnemonic()` is always the entry point for this entire use case / flow, then we're already guaranteed to have the cls set appropriately.

It might even be preferable for it to remain `None` outside of an active pending mnemonic flow so that it'll throw exceptions if future code tries to use any of these methods at a nonsense / uninitialized point.